src/app/locales/vi_VN.ts

```diff
@@ -0,0 +1,513 @@
+  },
+  'dependencies update': {
+    action:
+      'Nếu như dự án của bạn phụ thuộc vào các thư viện Angular khác, chúng tôi khuyến cáo bạn nên cập nhật lên phiên bản mới nhất của những thư viện này. Ở một số trường hợp, việc cập nhật phiên bản cho các thư viện bên thứ 3 là cần thiết để có thể giải quyết các vấn đề không tương thích về APIs. Tham khảo `ng update` hoặc `npm outdated` để biết thêm về những thư viện nào cần được cập nhật.'
+  },
+  'ivy update': {
+    action:
+      'Trong quá trình nâng cấp lên phiên bản 9, dự án của bạn đã được chuyển hoá (khi cần) thông qua việc thay đổi code tự động. Việc này dùng để xoá bỏ những API không còn tương thích hoặc đã bị khai tử ra khỏi code của bạn. Bạn có thể xem qua lại các thay đổi này, và tham khảo [Updating to version 9 guide](https://v9.angular.io/guide/updating-to-version-9) để nắm rõ hơn về những sự thay đổi này.'
+  },
+  stylesUpdate: {
+    action:
+      'Những CSS styles hoặc classes ở những phiên bản trước được áp dụng với hình thức là "cái cuối cùng sẽ thắng", nhưng bây giờ các CSS styles và classes này sẽ được áp dụng với một thứ tự ưu tiên đã được định sẵn. Đọc thêm tại [Styling Precedence](https://angular.io/guide/template-syntax#styling-precedence).'
+  },
+  ModuleWithProviders: {
+    action:
+      'Nếu bạn là tác giả của một thư viện dành cho Angular và bạn có viết hàm mà trả về `ModuleWithProviders` (ví dụ điển hình là hàm `forRoot()` thường gặp), bạn cần phải khai báo kiểu Generics cho `ModuleWithProviders`. Đọc thêm tại [angular.io](https://v9.angular.io/guide/deprecations#modulewithproviders-type-without-a-generic)'
+  },
+  wtf: {
+    action:
+      'Hỗ trợ cho Web Tracing Framework trong Angular đã bị khai tử ở phiên bản 8. Bạn nên dừng việc sử dụng bất cứ APIs nào của `wtf*`. Để kiểm tra hiệu năng, chúng tôi khuyến cáo bạn sử dụng [browser performance tools](https://developers.google.com/web/tools/lighthouse/audits/user-timing).'
+  },
+  es5browser: {
+    action:
+      'Xoá bỏ bất cứ `es5BrowserSupport` nào có trong `angular.json` của bạn và đặt giá trị cho `target` là `es2015` trong `tsconfig.json`. Angular từ giờ sẽ sử dụng `browserlist` để xác định xem build ES5 có cần thiết hay không. `ng update` sẽ thực hiện việc thay đổi này tự động.'
+  },
+  'ngForm selector': {
+    action: 'Nếu bạn sử dụng `ngForm` để tạo Angular Forms, bạn nên thay thế bằng `ng-form`.',
+  },
+  'typings compilation': {
+    action:
+      'Chúng tôi đã cập nhật `tsconfig.app.json` để giới hạn những Files sẽ được biên dịch. Nếu bạn phụ thuộc vào việc có những Files khác cần được thêm vào trong quá trình biên dịch, ví dụ như `typings.d.ts`, bạn cần phải thêm File này vào quá trình biên dịch bằng tay.'
+  },
+  debug: {
+    action:
+      'Với việc Ivy giờ là bộ máy biểu diễn mô hình (rendering) mặc định trong Angular 9, đối với bất cứ vấn đề nào về việc tương thích, mời bạn đọc qua [Ivy compatibility guide](https://angular.io/guide/ivy-compatibility).'
```

Comment:
Ivy compatibility guide = "hướng dẫn tương thích Ivy"

Reply:
"Hướng dẫn tương thích Ivy" doesn't flow so well imho. I'll think about this more.